### PR TITLE
* #101: Added error builder.

### LINKED ISSUE #101: Use our new error builder to unify error handling

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #101

### CODE REVIEW

File: src/main/java/com/exasol/releasedroid/usecases/BaseRepository.java [hunk 1]
```diff
@@ -52,10 +55,11 @@ public String getSingleFileContentAsString(final String filePath) {
...
         } catch (final IOException exception) {
-            throw new RepositoryException(
-                    "E-REP-GH-2: Cannot find or read the file '" + filePath + "' in the repository "
-                            + this.ghRepository.getName() + ". Please add this file according to the User Guide.",
-                    exception);
+            throw new RepositoryException(ExaError.messageBuilder("E-REP-GH-2")
+                                                  .message("Cannot find or read the file {{filePath}} in the repository {{repositoryName}}.")
+                                                  .parameter("filePath", filePath) //
+                                                  .parameter("repositoryName", this.ghRepository.getName()) //
+                                                  .mitigation("Please add this file according to the User Guide.").toString(), exception);
```
Comment (morazow) [inline src/main/java/com/exasol/releasedroid/usecases/BaseRepository.java:62]:
```suggestion
                                                  .mitigation("Please add this file according to the user guide.").toString(), exception);
```

File: src/main/java/com/exasol/releasedroid/usecases/BaseRepository.java [hunk 2]
```diff
@@ -68,8 +72,11 @@ public void updateFileContent(final String filePath, final String newContent, fi
         try {
             this.getFileContent(filePath).update(newContent, commitMessage, this.branch.getName());
         } catch (final IOException exception) {
-            throw new RepositoryException("E-REP-GH-6: Cannot update the file '" + filePath + "' in the repository "
-                    + this.ghRepository.getName() + ".", exception);
+            throw new RepositoryException(ExaError.messageBuilder("E-REP-GH-6")
+                                                  .message("Cannot update the file {{filePath}} in the repository {{repositoryName}}.")
+                                                  .parameter("filePath", filePath) //
+                                                  .parameter("repositoryName", this.ghRepository.getName()) //
+                                                  .mitigation("Please add this file according to the User Guide.").toString(), exception);
```
Comment (morazow) [inline src/main/java/com/exasol/releasedroid/usecases/BaseRepository.java:79]:
```suggestion
                                                  .mitigation("Please add this file according to the user guide.").toString(), exception);
```

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/releasedroid/github/GithubAPIAdapter.java [hunk 3]
```diff
@@ -218,9 +231,10 @@ private void waitMinutes(final int minutes) {
 
     private void validateWorkflowConclusion(final String workflowConclusion) throws GitHubException {
         if (!workflowConclusion.equals("success")) {
-            throw new GitHubException(
-                    "E-RR-GH-1: Workflow run failed. Please check the action logs on the GitHub to analyze the problem. Run result: "
-                            + workflowConclusion);
+            throw new GitHubException(ExaError.messageBuilder("E-RR-GH-2")
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/github/GithubAPIAdapter.java:234]: Yes, I renamed it

File: src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java [hunk 4]
```diff
@@ -61,16 +63,17 @@ private Result validateExecutions(final List<PluginExecution> executions) {
...
+                .mitigation("PLease, check a user guide to add it.").toString());
     }
 
     private Result validateConfigurations(final PluginExecution execution) {
         final Object configuration = execution.getConfiguration();
         if ((configuration == null) || !configuration.toString().contains("--pinentry-mode")) {
-            return ValidationResult.failedValidation("E-RR-VAL-16",
-                    "The 'maven-gpg-plugin' misses configuration of the 'sign-artifacts' execution. "
-                            + "PLease, check the user guide to add it.");
+            return ValidationResult.failedValidation(ExaError.messageBuilder("E-RR-VAL-7")
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java:74]: 7 was missing

**Review by morazow — APPROVED** [review]

**Pull request merged**